### PR TITLE
wrap V4 background layers in 3D OLCS by using v4 background store

### PR DESCRIPTION
**Pull request by mki-c2c**
- [x] use raster fallback layer for 3d in OLCS
- [x] handle layer exclusion => https://jira.camptocamp.com/browse/GSLUX-644
- [ ] register styles correctly for raster layers in 3d => https://jira.camptocamp.com/browse/GSLUX-648
- [x] handle 3D layers in catalog -> https://jira.camptocamp.com/browse/GSLUX-642
- [x] chase all occurences of this.backgroundLayerMgr_ = ngeoBackgroundLayerMgr => https://jira.camptocamp.com/browse/GSLUX-645
- [ ] retrieve style reset button => https://jira.camptocamp.com/browse/GSLUX-646
- [ ] fix permalink => https://jira.camptocamp.com/browse/GSLUX-647
- [ ] sync medium style editor with style store => https://jira.camptocamp.com/browse/GSLUX-647

### CODE REVIEW

**Review by tkohr — APPROVED** [review]
Thanks for this huge work @mki-c2c! Overall looking good and working fine.

I propose we can merge on to move forward, and we will do the remaining to-dos in separate PRs.

File: geoportal/geoportailv3_geoportal/static-ngeo/js/apps/MainController.js [hunk 1]
```diff
@@ -560,11 +571,16 @@ const MainController = function(
...
+
+    const bgLayer = useOpenLayers().getLayerFromCache(this.mapStore_.bgLayer);
     const label = bgLayer.get('label');
     this.mediumStylingData = getDefaultMediumStyling(label); // start again from a fresh style
     const mediumStyles = this.mediumStylingData.filter(m => 'color' in m);
-    const mbMap =  bgLayer.getMapBoxMap();
+    const mbMap =  bgLayer.getMapLibreMap();
     for (let i = 0; i < selectedItem['colors'].length; ++i) {
       const item = mediumStyles[i];
       item.color = selectedItem['colors'][i];
```
[the remark this answers is not in the document]
Comment (mki-c2c) [reply geoportal/geoportailv3_geoportal/static-ngeo/js/apps/MainController.js:586]: probably just remove all the legacy code below

File: geoportal/geoportailv3_geoportal/static-ngeo/js/apps/MainController.js [hunk 2]
```diff
@@ -581,7 +597,17 @@ const MainController = function(
...
+    )
+    this.styleStore_.setStyle(item)
+    return
+
+    ////////////////////////////////////////////////////////////////////////
+
+    const bgLayer = useOpenLayers().getLayerFromCache(this.mapStore_.bgLayer);
     const mbMap =  bgLayer.getMapBoxMap();
     applyStyleFromItem(mbMap, item, bgLayer.get('label'));
     this.debouncedSaveStyle_();
```
[the remark this answers is not in the document]
Comment (mki-c2c) [reply geoportal/geoportailv3_geoportal/static-ngeo/js/apps/MainController.js:613]: probably just remove all the legacy code below

File: geoportal/geoportailv3_geoportal/static-ngeo/js/apps/MainController.js [hunk 3]
```diff
@@ -1413,7 +1445,12 @@ const MainController = function(
...
-    const bgLayer = this.backgroundLayerMgr_.get(this.map);
+    this.styleStore_.setStyle(null)
+    return
+
+    /////////////////////////////////////////////////////////////////////////
+
+    const bgLayer = useOpenLayers().getLayerFromCache(this.mapStore_.bgLayer);
     this.appMvtStylingService.removeStyles(bgLayer);
     bgLayer.getMapBoxMap().setStyle(bgLayer.get('defaultMapBoxStyle'));
     this.mediumStylingData = getDefaultMediumStyling(bgLayer.get('label'));
```
[the remark this answers is not in the document]
Comment (mki-c2c) [reply geoportal/geoportailv3_geoportal/static-ngeo/js/apps/MainController.js:1456]: probably just remove all the legacy code below

File: geoportal/geoportailv3_geoportal/static-ngeo/js/apps/MainController.js [hunk 4]
```diff
@@ -1485,11 +1522,11 @@ MainController.prototype.getUrlVtStyle = function() {
...
-  if (!active) {
-    this.appMvtStylingService.unpublishIfSerial(this.map_);
-    return;
-  }
-  this.appMvtStylingService.publishIfSerial(this.map_);
+  // if (!active) {
+  //   this.appMvtStylingService.unpublishIfSerial(this.map_);
+  //   return;
+  // }
+  // this.appMvtStylingService.publishIfSerial(this.map_);
```
[the remark this answers is not in the document]
Comment (mki-c2c) [reply geoportal/geoportailv3_geoportal/static-ngeo/js/apps/MainController.js:1529]: I guess nothing to do here

**Comment by mki-c2c** [comment]
hello @tkohr you are right, we should just remove the dead code.
Just remove the lines which you highlighted, and those after the return statements. I was kind of afraid, but I see no need to keep them. We can always come back in git history if needed.

**Comment by tkohr** [comment]
Ok, I just commented the dead code for now to keep it until 3D is completely done.

The v4 dependency now points on latest commit, so I think we are ready to merge.

**Pull request merged**